src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -806,4 +806,40 @@ class Dockerfile extends DefaultTask {
             "LABEL"
         }
     }
+
+    /**
+     * Helper Instruction used by DockerJavaApplicationPlugin
+     * to allow customizing generated ENTRYPOINT/CMD
+     */
+    static class CompositeExecInstruction implements Instruction {
+        @Internal
+        private List<Instruction> instructions = new ArrayList<Instruction>()
```

Comment:
Right now this is still exposed and could be used for other instruction types.  It could certainly be locked down to at least be read-only except for the defined methods or the list could be swapped out for specific nullable references to each of the supported commands but this seemed like a good place to start.